crates/nu-cli/src/completions/command_completions.rs

```diff
@@ -159,179 +152,3 @@ impl CommandCompletion {
-        };
-
-        if !subcommands.is_empty() {
-            return subcommands;
-        }
-
-        let config = working_set.get_config();
-        if matches!(self.flat_shape, nu_parser::FlatShape::External)
-            || matches!(self.flat_shape, nu_parser::FlatShape::InternalCall(_))
-            || ((span.end - span.start) == 0)
-            || is_passthrough_command(working_set.delta.get_file_contents())
-        {
-            // we're in a gap or at a command
-            if working_set.get_span_contents(span).is_empty() && !self.force_completion_after_space
-            {
-                return vec![];
-            }
-            self.complete_commands(
-                working_set,
-                span,
-                offset,
-                config.completions.external.enable,
-                options,
-            )
-        } else {
-            vec![]
-        }
-    }
-}
-
-pub fn find_non_whitespace_index(contents: &[u8], start: usize) -> usize {
-    match contents.get(start..) {
-        Some(contents) => {
-            contents
-                .iter()
-                .take_while(|x| x.is_ascii_whitespace())
-                .count()
-                + start
-        }
-        None => start,
-    }
-}
-
-pub fn is_passthrough_command(working_set_file_contents: &[CachedFile]) -> bool {
-    for cached_file in working_set_file_contents {
-        let contents = &cached_file.content;
-        let last_pipe_pos_rev = contents.iter().rev().position(|x| x == &b'|');
-        let last_pipe_pos = last_pipe_pos_rev.map(|x| contents.len() - x).unwrap_or(0);
-
-        let cur_pos = find_non_whitespace_index(contents, last_pipe_pos);
-
-        let result = match contents.get(cur_pos..) {
-            Some(contents) => contents.starts_with(b"sudo ") || contents.starts_with(b"doas "),
-            None => false,
-        };
-        if result {
-            return true;
-        }
-    }
-    false
-}
-
-#[cfg(test)]
-mod command_completions_tests {
-    use super::*;
-    use nu_protocol::engine::EngineState;
-    use std::sync::Arc;
-
-    #[test]
-    fn test_find_non_whitespace_index() {
-        let commands = [
-            ("    hello", 4),
-            ("sudo ", 0),
-            (" 	sudo ", 2),
-            ("	 sudo ", 2),
-            ("	hello ", 1),
-            ("	  hello ", 3),
-            ("    hello | sudo ", 4),
-            ("     sudo|sudo", 5),
-            ("sudo | sudo ", 0),
-            ("	hello sud", 1),
-        ];
-        for (idx, ele) in commands.iter().enumerate() {
-            let index = find_non_whitespace_index(ele.0.as_bytes(), 0);
-            assert_eq!(index, ele.1, "Failed on index {}", idx);
-        }
-    }
-
-    #[test]
-    fn test_is_last_command_passthrough() {
-        let commands = [
-            ("    hello", false),
-            ("    sudo ", true),
-            ("sudo ", true),
-            ("	hello", false),
-            ("	sudo", false),
-            ("	sudo ", true),
```

Comment:
Not really up to date, in which direction are we going with the completions behind `sudo` and friends?

Reply:
`sudo` should behave the exact same way as before, i.e. the immediate argument after it is completed as internals + externals (not quite accurate with flags). Otherwise resort to external completer.

This test cases are obsolete because the helper function is no longer required, the check is done in `completer.rs` with good accuracy.

Reply:
Yeah dropping the tests if its just overexercising some utility instead of the thing we care about makes sense.

Generally agree with the vibe that the things for `sudo` and friends are hacky :D